syne_tune/optimizer/schedulers/multiobjective/nsga2_searcher.py

```diff
@@ -0,0 +1,177 @@
+            self._mode = [mode] * len(metric)
+        else:
+            self._mode = mode
+
+        self.hp_names = []
+        for hp_name, hp in config_space.items():
+            if isinstance(hp, Domain):
+                self.hp_names.append(hp_name)
+                assert type(hp) in [
+                    Categorical,
+                    Integer,
+                    Float,
+                    FiniteRange,
+                ], f"Type {type(hp)} for hyperparameter {hp_name} not supported."
+
+        self.problem = MultiObjectiveMixedVariableProblem(
+            config_space=config_space, n_var=len(self.hp_names), n_obj=len(metric)
+        )
+        self.algorithm = NSGA2(
+            pop_size=population_size,
+            sampling=MixedVariableSampling(),
+            mating=MixedVariableMating(
+                eliminate_duplicates=MixedVariableDuplicateElimination()
+            ),
+            eliminate_duplicates=MixedVariableDuplicateElimination(),
+        )
+        self.algorithm.setup(
+            problem=self.problem, termination=("n_eval", 2**32 - 1), verbose=False
+        )
+
+        self.current_population = self.algorithm.ask()
+        self.current_individual = 0
+        self.observed_values = []
+
+    def _update(self, trial_id: str, config: dict, result: dict):
+        observed_metrics = {}
+        for mode, metric in zip(self._mode, self._metric):
+            value = result[metric]
+            if mode == "max":
+                value *= -1
+            observed_metrics[metric] = value
+
+        self.observed_values.append(list(observed_metrics.values()))
+
+        if len(self.observed_values) == len(self.current_population):
+            static = StaticProblem(self.problem, F=np.array(self.observed_values))
+            Evaluator().eval(static, self.current_population)
+            # self.algorithm.evaluator.eval(self.problem, self.current_population)
+            self.algorithm.tell(infills=self.current_population)
+
+            self.current_population = self.algorithm.ask()
+            self.observed_values = []
+            self.current_individual = 0
+
+    def get_config(self, **kwargs) -> Optional[Dict[str, Any]]:
+        """Suggest a new configuration.
+
+        Note: Query :meth:`_next_initial_config` for initial configs to return
+        first.
+
+        :param kwargs: Extra information may be passed from scheduler to
+            searcher
+        :return: New configuration. The searcher may return None if a new
+            configuration cannot be suggested. In this case, the tuning will
+            stop. This happens if searchers never suggest the same config more
+            than once, and all configs in the (finite) search space are
+            exhausted.
+        """
+
+        if self.current_individual >= len(self.current_population):
+            raise Exception(
+                "It seems that some configurations are sill pending, while querying a new configuration."
```

Comment:
This is a bit sad. Isn't there an easy way to make this work?

Reply:
I guess the best option would be to have a synchronous FIFO scheduler. We could implement some hack in there, but that would lead to a different algorithm and bias the results